src/common.ts

```diff
@@ -20,16 +20,21 @@ export class GaxiosError<T = any> extends Error {
   code?: string;
   response?: GaxiosResponse<T>;
   config: GaxiosOptions;
+  status: Number;
   constructor(
     message: string,
     options: GaxiosOptions,
-    response: GaxiosResponse<T>
+    response: GaxiosResponse<T>,
+    error?: Error | NodeJS.ErrnoException
```

Comment:
Lastly, we can expose the provided error if someone wants to debug further
```suggestion
    public error?: Error | NodeJS.ErrnoException
```